src/Config.h

```diff
@@ -23,11 +25,16 @@ class Callback {
   virtual void operator()(string const &msg) = 0;
 };
 
-struct Listener_impl;
+struct Listener_impl{
+  std::unique_ptr<KafkaW::BaseConsumer> consumer;
+  std::mutex mx;
+  std::condition_variable cv;
+  int connected = 0;
```

Comment:
Change `connected` to a `bool`, this allows some simplification in the `.cxx` file too.